lib/dot-compat.js

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+        }
+    }
+
+    // copy over linterOptions
+    for (const key of linterOptionsKeysToCopy) {
+        if (key in eslintrcConfig && typeof eslintrcConfig[key] !== "undefined") {
+            dotConfig.linterOptions = linterOptions;
+            linterOptions[key] = eslintrcConfig[key];
+        }
+    }
+
+    // move ecmaVersion a level up
+    if (languageOptions.parserOptions) {
+
+        if ("ecmaVersion" in languageOptions.parserOptions) {
+            languageOptions.ecmaVersion = languageOptions.parserOptions.ecmaVersion;
+            delete languageOptions.parserOptions.ecmaVersion;
+        }
+
+        if ("sourceType" in languageOptions.parserOptions) {
+            languageOptions.sourceType = languageOptions.parserOptions.sourceType;
+            delete languageOptions.parserOptions.sourceType;
+        }
+
+        // check to see if we even need parserOptions anymore and remove it if not
+        if (Object.keys(languageOptions.parserOptions).length === 0) {
+            delete languageOptions.parserOptions;
+        }
+    }
+
+    if (eslintrcConfig.ignorePattern) {
+        configs.unshift({
+            ignores: eslintrcConfig.ignorePattern.patterns
+        });
+    }
+
+    // overrides
+    if (eslintrcConfig.criteria) {
+
+        for (const { includes, excludes } of eslintrcConfig.criteria.patterns) {
+            if (includes) {
+                dotConfig.files = includes.map(include => include.pattern);
+            }
+
+            if (excludes) {
+                if (!includes) {
+                    dotConfig.files = [excludes.map(exclude => `!${exclude.pattern}`)];
+                } else {
+                    dotConfig.ignores = excludes.map(exclude => exclude.pattern);
+                }
+            }
+
+        }
+
+    }
+
+    // translate plugins
+    if (eslintrcConfig.plugins && typeof eslintrcConfig.plugins === "object") {
+        debug(`Translating plugins: ${eslintrcConfig.plugins}`);
+
+        dotConfig.plugins = {};
+
+        for (const pluginName of Object.keys(eslintrcConfig.plugins)) {
+
+            debug(`Translating plugin: ${pluginName}`);
+            debug(`Resolving plugin '${pluginName} relative to ${resolvePluginsRelativeTo}`);
+
+            const plugin = eslintrcConfig.plugins[pluginName].definition;
+
+            dotConfig.plugins[pluginName] = plugin;
+
+            // create a config for any processors
+            if (plugin.processors) {
+                for (const processorName of Object.keys(plugin.processors)) {
+                    if (processorName.startsWith(".")) {
+                        debug(`Assigning processor: ${pluginName}/${processorName}`);
+
+                        configs.unshift({
+                            files: [`**/*${processorName}`],
+                            processor: plugin.processors[processorName]
+                        });
+                    }
+
+                }
+            }
```

Comment:
This is not necessary. `ConfigArrayFactory` has created it already: [/lib/config-array-factory.js#L1063-L1091](https://github.com/eslint/eslintrc/blob/f30bb4935aaf3f4c1b268490da495a59647e58d8/lib/config-array-factory.js#L1063-L1091)

Reply:
Ah nice, thanks!

Reply:
It turns out that the implicit-processor config type still specifies the processor as a string that requires a lookup. Given that, my logic here handles it in a way that's easier for me to manage.